### PR TITLE
More thoroughly search for possible wesnothd executable names

### PR DESCRIPTION
**Pull request by Pentarctagon**

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
This kinda seems wrong. If you have wesnothd and wesnothd-devel, this'll make wesnoth-devel invoke wesnothd when it should probably invoke wesnothd-devel instead.

File: src/filesystem.cpp [hunk 1]
```diff
@@ -1064,43 +1064,31 @@ std::string get_exe_dir()
...
-			return wesnothd;
-		}
-	}
-
-	return "";
+	auto [exe_dir, exe] = get_exe_dir();
+	// macOS doesn't call the wesnoth client executable "wesnoth"
+	// otherwise, add any suffix after the "wesnoth" part of the executable name to wesnothd's name
+#ifdef __APPLE__
+	return exe_dir + "/" + get_program_invocation("wesnothd");
```
Comment (CelticMinstrel) [inline src/filesystem.cpp:1088]: Is this right? What does `get_program_invocation` do?

Comment (Pentarctagon) [reply src/filesystem.cpp:1088]: If built in Debug mode, append `-debug`. If Windows, also append `.exe`.

Comment (CelticMinstrel) [reply src/filesystem.cpp:1088]: I don't think Xcode appends `-debug`.

Comment (Pentarctagon) [reply src/filesystem.cpp:1088]: This PR changes it to only append `-debug` if it's Windows, but I'm not sure if that's needed at all. cmake never did that, scons used to but doesn't anymore, and Iris confirmed that xcode/macOS doesn't either. So it's just a question of whether the cmake-generated Visual Studio project files still do it or not.

Comment (soliton-) [reply src/filesystem.cpp:1088]: Would be better if this used an Xcode specific define.

Alternatively no special logic and check if the file exists and have a fixed "wesnothd" fallback.

File: src/filesystem.cpp [hunk 2]
```diff
@@ -1033,7 +1033,7 @@ bool set_cwd(const std::string& dir)
 	return true;
 }
 
-std::string get_exe_dir()
+std::pair<std::string, std::string> get_exe_dir()
```
Comment (soliton-) [inline src/filesystem.cpp:1036]: This seems odd. I'd rather add a `get_exe` or `get_exe_path` function. `get_exe_dir` can then just use that function and return only the dir.

File: src/preferences/general.cpp [hunk 3]
```diff
@@ -240,6 +240,8 @@ config* get_prefs(){
 void load_base_prefs() {
 	try{
 #ifdef DEFAULT_PREFS_PATH
+		// NOTE: the system preferences file is only ever relevant for the first time wesnoth starts
+		//       any default values will subsequently be written to the normal preferences file, which takes precedence over any values in the system preferences file
```
Comment (soliton-) [inline src/preferences/general.cpp:244]: This does not seem to belong in this PR.

**Review by soliton- — APPROVED** [review]
Looks good to me. The default prefs comment belongs somewhere else though.

**Comment by Pentarctagon** [comment]
I'm not making another PR for a comment.

**Comment by soliton-** [comment]
No reason to make a PR at all. If you must leave it in this PR (no idea why) then make it a separate commit.
